transbank/webpay/webpayplus/transaction.py

```diff
@@ -0,0 +1,161 @@
+            "Content-Type": "application/json",
+        })
+
+        http_client = WebpayPlus.http_client
+        final_url = base_url + cls.COMMIT_TRANSACTION_ENDPOINT + "/" + token_ws
+        http_response = http_client.put(final_url, headers=headers)
+
+        response_json = http_response.json()
+
+        if response_json.get('error_message') is not None:
+            raise Exception(response_json["error_message"])
+
+        json_data = response_json
+        transaction_commit_response = TransactionCommitResponse(json_data)
+        return transaction_commit_response
+
+    @classmethod
+    def refund(cls, token, amount, options=None):
+        commerce_code = WebpayPlus.commerce_code()
+        api_key = WebpayPlus.api_key()
+        base_url = WebpayPlus.integration_type_url()
+
+        if options is not None:
+            commerce_code = options.commerce_code
+            api_key = options.api_key
+            base_url = WebpayPlus.integration_type_url(options.integration_type)
+
+        headers = dict({
+            "Tbk-Api-Key-Id": commerce_code,
+            "Tbk-Api-Key-Secret": api_key,
+            "Content-Type": "application/json",
+        })
+
+        payload = json.dumps(dict({
+            "amount": amount
+        }))
+
+        http_client = WebpayPlus.http_client
+        final_url = base_url + cls.REFUND_TRANSACTION_ENDPOINT.format(token)
+
+        http_response = http_client.post(final_url, data=payload, headers=headers)
+
+        if (http_response.status_code < 200) or (http_response.status_code > 300) or http_response is None:
+            raise Exception('Could not obtain a response from the service', -1)
+        response_json = http_response.json()
+
+        if response_json.get('error_message') is not None:
+            raise Exception(response_json["error_message"])
+
+        json_data = response_json
+        transaction_refund_response = TransactionRefundResponse(json_data)
+        return transaction_refund_response
+
+    @classmethod
+    def status(cls, token, options=None):
+        commerce_code = WebpayPlus.commerce_code()
+        api_key = WebpayPlus.api_key()
+        base_url = WebpayPlus.integration_type_url()
+
+        if options is not None:
+            commerce_code = options.commerce_code
+            api_key = options.api_key
+            base_url = WebpayPlus.integration_type_url(options.integration_type)
+
+        headers = dict({
+            "Tbk-Api-Key-Id": commerce_code,
+            "Tbk-Api-Key-Secret": api_key,
+            "Content-Type": "application/json",
+        })
+
+        http_client = WebpayPlus.http_client
+        final_url = base_url + cls.GET_TRANSACTION_STATUS_ENDPOINT.format(token)
+
+        http_response = http_client.get(final_url, headers=headers)
+
+        if (http_response.status_code < 200) or (http_response.status_code > 300) or http_response is None:
+            raise Exception('Could not obtain a response from the service', -1)
```

Comment:
Why 3xx codes are being handled here? Are we expecting redirections? Are we not following redirections?

Also why are 4xx and 5xx codes just raising exceptions without inspecting the body? Isn't the 'error_message' going to be useful when we get a 5xx of 4xx error?

Reply:
Of course @leosoto you've got the reason, we don't need to handle redirections. At this case I must to change it in every code section. About to raise exceptions, I must change it to return the http code in any case (4xx or 5xx error codes)

Reply:
This requires a complex refactor, a solution more scalable, hence those changes won't working here, but in other refactor branch